From: Dotan Barak <dotanb@dev.mellanox.co.il>
Date: Sun, 21 Apr 2013 15:10:00 +0000
Subject: IB/mlx4: Disable VLAN stripping for RAW PACKET QPs
Patch-mainline: v3.10-rc1
Git-commit: 02d7ef6f9dfdd0756441913e34008e8883195deb
References: bug#919382 FATE#317529

Fix the asymmetric behavior w.r.t VLAN insertion/stripping for RAW
PACKET QPs -- we don't insert on send and need not strip on receive.

Signed-off-by: Dotan Barak <dotanb@dev.mellanox.co.il>
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/infiniband/hw/mlx4/qp.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/infiniband/hw/mlx4/qp.c
+++ b/drivers/infiniband/hw/mlx4/qp.c
@@ -1278,6 +1278,8 @@ static int __mlx4_ib_modify_qp(struct ib
 		context->sq_size_stride |= !!qp->sq_no_prefetch << 7;
 		if (ibqp->qp_type == IB_QPT_XRC)
 			context->xrcd = cpu_to_be32((u32) qp->xrcdn);
+		if (ibqp->qp_type == IB_QPT_RAW_PACKET)
+			context->param3 |= cpu_to_be32(1 << 30);
 	}
 
 	if (qp->ibqp.uobject)
